experimentation/src/main/java/com/automattic/android/experimentation/local/FileBasedCache.kt

```diff
@@ -0,0 +1,50 @@
+package com.automattic.android.experimentation.local
+
+import com.automattic.android.experimentation.domain.Assignments
+import com.automattic.android.experimentation.remote.AssignmentsDtoJsonAdapter
+import com.automattic.android.experimentation.remote.AssignmentsDtoMapper.toAssignments
+import com.automattic.android.experimentation.remote.AssignmentsDtoMapper.toDto
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+
+internal class FileBasedCache(
+    private val cacheDir: File,
+    private val moshi: Moshi = Moshi.Builder().build(),
+    private val jsonAdapter: AssignmentsDtoJsonAdapter = AssignmentsDtoJsonAdapter(moshi),
+) {
+
+    private val assignmentsFile = File(cacheDir, "assignments.json")
+    private val type = Types.newParameterizedType(
+        Map::class.java,
+        Long::class.javaObjectType,
+        String::class.java,
+    )
+    private val wrapperAdapter = moshi.adapter<Map<Long, String>>(type)
+
+    suspend fun getAssignments(): Assignments? {
+        return withContext(Dispatchers.IO) {
+            assignmentsFile.takeIf { it.exists() }?.readText()?.let { json ->
+                val fromJson = wrapperAdapter.fromJson(json) ?: return@let null
+
+                val (fetchedAt, dtoJson) = fromJson.toList().first()
+
+                val dto = jsonAdapter.fromJson(dtoJson)
+
+                dto?.toAssignments(fetchedAt)
+            }
+        }
+    }
+
+    suspend fun saveAssignments(assignments: Assignments) {
```

Comment:
Question (❓): On the [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android) side, the saving implementation is based on `AssignmentsModel` and not `Assignments` ([storeFetchedAssignments(...)](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/3f240ecc9161efca1d75183a29640bd21db95cb6/fluxc/src/main/java/org/wordpress/android/fluxc/store/ExperimentStore.kt#L43C17-L46)), which means that there is no need to have this `dto` + `fetchedAt` pair logic, nor lose the `control` information when mapping the `Variation.Control -> null`, as this `Variation.Control` can either mean `null` or `control`. Wdyt? 🤔

Reply:
My reasoning here is that remote layer shouldn't communicate directly with persistence. 

This rule is already broken in 98b9674f84f5e66acd8b1db14a9fe3eeac104e71 by not introducing a persistence-only model, but I did it consciously to not introduce a new, twin one.

I'd like to keep the unidirectional control flow: remote -> domain -> persistence though, hence using `Assigmnets` domain model here.

Reply:
Sold, I then agree with your reasoning of using `Assignments` domain model, even if makes it a bit more complicated, thanks for the clarification! 💯